contrib/mypy/src/python/pants/contrib/mypy/tasks/mypy_task.py

```diff
@@ -13,19 +14,39 @@
 from pants.base.build_environment import get_buildroot
 from pants.base.exceptions import TaskError
 from pants.base.workunit import WorkUnit, WorkUnitLabel
+from pants.build_graph.target import Target
+from pants.task.lint_task_mixin import LintTaskMixin
 from pants.util.contextutil import temporary_file_path
 from pants.util.memo import memoized_property
 from pex.interpreter import PythonInterpreter
 from pex.pex import PEX
 from pex.pex_info import PexInfo
 
 
-class MypyTask(ResolveRequirementsTaskBase):
-  """Invoke the mypy static type analyzer for Python."""
+class MypyTaskError(TaskError):
+  """Indicates a TaskError from a failing MyPy run."""
+
+
+class MypyTask(LintTaskMixin, ResolveRequirementsTaskBase):
+  """Invoke the mypy static type analyzer for Python.
+
+  Mypy lint task filters out target_roots that are not properly tagged according to
+  --whitelisted-tag-name (defaults to None, and no filtering occurs if this option is 'None'),
+  and executes MyPy on targets in context from whitelisted target roots.
+  Next, if any transitive targets from the filtered roots are not whitelisted, a warning
+  will be printed.
+
+  'In context' meaning in the sub-graph where a whitelisted target is the root
+  """
 
   _MYPY_COMPATIBLE_INTERPETER_CONSTRAINT = '>=3.5'
   _PYTHON_SOURCE_EXTENSION = '.py'
 
+  deprecated_options_scope = 'mypy'
+  deprecated_options_scope_removal_version = '1.20.0.dev2'
```

Comment:
This needs to be later now that 1.20.0.dev2 was cut, I believe. Think this should be 1.21.0.dev0? cc @wisechengyi.